xarray/core/coordinates.py

```diff
@@ -312,7 +316,14 @@ def to_dataset(self) -> "Dataset":
         return Dataset._construct_direct(coords, set(coords))
 
     def __delitem__(self, key: Hashable) -> None:
-        del self._data._coords[key]
+        if key in self:
+            del self._data._coords[key]
+            if key in self._data.indexes:
+                indexes = dict(self._data.indexes)
+                indexes.pop(key)
+                self._data._indexes = indexes
```

Comment:
`_indexes` should be unique to each `DataArray` object, so it's fine to update this in-place like what we do on `Dataset.__delitem__`:
```suggestion
                del self._data._indexes[key]
```

Reply:
:+1: